AsyncDisplayKit/Layout/ASLayout.h

```diff
@@ -0,0 +1,46 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#import <UIKit/UIKit.h>
+#import <AsyncDisplayKit/ASAssert.h>
+#import <AsyncDisplayKit/ASLayoutable.h>
+
+/** Represents the computed size of a layout node, as well as the computed sizes and positions of its children. */
+@interface ASLayout : NSObject
+
+@property (nonatomic, readonly) id<ASLayoutable> layoutableObject;
+@property (nonatomic, readonly) CGSize size;
+/** 
+ * Each item is of type ASLayoutChild. 
+ */
+@property (nonatomic, readonly) NSArray *children;
+
++ (instancetype)newWithLayoutableObject:(id<ASLayoutable>)layoutableObject
+                                   size:(CGSize)size
+                               children:(NSArray *)children;
+
+/**
+ * Convenience that does not have any children.
+ */
++ (instancetype)newWithLayoutableObject:(id<ASLayoutable>)layoutableObject size:(CGSize)size;
+
+@end
+
+@interface ASLayoutChild : NSObject
+
+@property (nonatomic, readonly) CGPoint position;
```

Comment:
Let me know what you think about removing ASLayoutChild completely, and adding a .position property to ASLayout itself.  What bad / weird effects would this cause?

I think the main issue is that the .position would be unset / invalid for the top level ASLayout (e.g. for a given node, its ASLayoutSpec could generate an ASLayout that has a valid size and valid "sublayouts" with both size and position, but not a valid position for the top-level ASLayout).  I think this would be OK, but I'm not sure if we need a special constant-type CGPoint to mark this as invalid (sorta like kCGRectNull; is there a similar thing for a CGPoint?)

Not as critical, but I'm also thinking it would be ideal to rename usages of "children:" that came from ComponentKit.  This term isn't used anywhere else in ASDK and it would be better to come up with a more specific term.  subLayouts, subLayoutable (eh, not really great), subnodes would be perfect but does this ever need to contain sub-elements that aren't actual nodes?  I guess that only applies to ASLayoutSpec, and the actual generated ASLayout should only ever describe the qualities of real subnodes, right?  If that's true, let's use subnodeLayouts: